x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/new_index/select_connector/connector_checkable.tsx

```diff
@@ -5,128 +5,195 @@
+                {isDisabled ? (
+                  <EuiText color="disabledText" size="xs">
+                    <h3>{name}</h3>
+                  </EuiText>
+                ) : (
+                  <EuiTitle size="xs">
+                    <h2>{name}</h2>
+                  </EuiTitle>
+                )}
+              </EuiFlexItem>
+              {isDisabled && (
+                <EuiFlexItem grow={false}>
+                  <PlatinumLicensePopover
+                    button={
+                      <EuiButtonIcon
+                        aria-label={i18n.translate(
+                          'xpack.enterpriseSearch.content.newIndex.selectConnector.openPopoverLabel',
+                          {
+                            defaultMessage: 'Open licensing popover',
+                          }
+                        )}
+                        iconType="questionInCircle"
+                        onClick={() => setIsLicensePopoverOpen(!isLicensePopoverOpen)}
+                      />
+                    }
+                    closePopover={() => setIsLicensePopoverOpen(false)}
+                    isPopoverOpen={isLicensePopoverOpen}
+                  />
+                </EuiFlexItem>
+              )}
+            </EuiFlexGroup>
           </EuiFlexItem>
-          {disabled && (
+          {showNativeBadge && (
             <EuiFlexItem grow={false}>
-              <PlatinumLicensePopover
+              <EuiPopover
                 button={
                   <EuiButtonIcon
-                    aria-label={i18n.translate(
-                      'xpack.enterpriseSearch.content.newIndex.selectConnector.openPopoverLabel',
-                      {
-                        defaultMessage: 'Open licensing popover',
-                      }
-                    )}
-                    iconType="questionInCircle"
-                    onClick={() => setIsPopoverOpen(!isPopoverOpen)}
+                    display="base"
+                    isDisabled={isDisabled}
+                    color="primary"
+                    iconType="boxesHorizontal"
+                    onClick={(e: MouseEvent) => {
+                      e.stopPropagation();
+                      e.preventDefault();
+                      setIsNativePopoverOpen(true);
+                    }}
                   />
                 }
-                closePopover={() => setIsPopoverOpen(false)}
-                isPopoverOpen={isPopoverOpen}
-              />
+                isOpen={isNativePopoverOpen}
+                closePopover={() => {
+                  setIsNativePopoverOpen(false);
+                }}
+              >
+                <EuiContextMenuPanel
+                  size="xs"
+                  items={[
+                    <EuiContextMenuItem
+                      key="native"
+                      onClick={(e) => {
+                        e.stopPropagation();
+                        onConnectorSelect('connector_type=native');
```

Comment:
Doing this with query params directly seems very error prone. I'd rather call this with a boolean and then determine what the query params should be in the parent component.